internal/executor/instance/persistentworker/remoteagent/remoteagent.go

```diff
@@ -148,3 +161,39 @@ func WaitForAgent(
 
 	return nil
 }
+
+func forwardViaSSH(cli *ssh.Client, logger logger.Lightweight, endpoint string) (string, error) {
+	// Expose local RPC service to the VM via SSH port forwarding
+	vmListener, err := cli.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return "", fmt.Errorf("%w: failed to set-up SSH port forwarding: %v", ErrFailed, err)
+	}
+
+	go func() {
+		for {
+			vmConn, err := vmListener.Accept()
+			if err != nil {
+				logger.Debugf("failed to accept connection from the VM on forwarded port: %v", err)
```

Comment:
Is there a need to wait a bit? 🤔

Reply:
> Is there a need to wait a bit? 🤔

The `Accept()` is blocking, so it should be fine.

Anyway, thanks for noticing this, I've realized we don't handle `io.EOF` here and fixed this in https://github.com/cirruslabs/cirrus-cli/pull/495/commits/3b2def425fa79e46ca1455c88c668e007ffc1e72.